packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,345 @@
+    function _setProxies() internal {
+        bytes memory code = vm.getDeployedCode("Proxy.sol:Proxy");
+        uint160 prefix = uint160(0x420) << 148;
+        for (uint256 i = 0; i < PROXY_COUNT; i++) {
+            address addr = address(prefix | uint160(i));
+            if (_notProxied(addr)) {
+                continue;
+            }
+
+            vm.etch(addr, code);
+            vm.store(addr, PROXY_ADMIN_ADDRESS, bytes32(uint256(uint160(Predeploys.PROXY_ADMIN))));
+
+            if (_isDefinedPredeploy(addr)) {
+                address implementation = _predeployToCodeNamespace(addr);
+                vm.store(addr, PROXY_IMPLEMENTATION_ADDRESS, bytes32(uint256(uint160(implementation))));
+            }
+        }
+    }
+
+    /// @notice LEGACY_ERC20_ETH is not being predeployed
+    function _setImplementations() internal {
+        _setLegacyMessagePasser();
+        _setDeployerWhitelist();
+        _setWETH9();
+        _setL2StandardBridge();
+        _setL2CrossDomainMessenger();
+        _setSequencerFeeVault();
+        _setOptimismMintableERC20Factory();
+        _setL1BlockNumber();
+        _setGasPriceOracle();
+        _setGovernanceToken();
+        _setL1Block();
+    }
+
+    function _setLegacyMessagePasser() internal {
+        address impl = _predeployToCodeNamespace(Predeploys.LEGACY_MESSAGE_PASSER);
+        vm.etch(impl, vm.getDeployedCode("LegacyMessagePasser.sol:LegacyMessagePasser"));
+    }
+
+    function _setDeployerWhitelist() internal {
+        address impl = _predeployToCodeNamespace(Predeploys.DEPLOYER_WHITELIST);
+        vm.etch(impl, vm.getDeployedCode("DeployerWhitelist.sol:DeployerWhitelist"));
+    }
+
+    /// @dev Contract metadata appended to deployed bytecode will differ
+    ///      from previous L2 genesis output
+    function _setWETH9() internal {
+        vm.etch(
+            Predeploys.WETH9,
+            vm.getDeployedCode("WETH9.sol:WETH9")
+        );
+        vm.store(
+            Predeploys.WETH9,
+            // string public name
+            hex"0000000000000000000000000000000000000000000000000000000000000000",
+            // "Wrapped Ether"
+            hex"577261707065642045746865720000000000000000000000000000000000001a"
+        );
+        vm.store(
+            Predeploys.WETH9,
+            // string public symbol
+            hex"0000000000000000000000000000000000000000000000000000000000000001",
+            // "WETH"
+            hex"5745544800000000000000000000000000000000000000000000000000000008"
+        );
+        vm.store(
+            Predeploys.WETH9,
+            // uint8 public decimals
+            hex"0000000000000000000000000000000000000000000000000000000000000002",
+            // 18
+            hex"0000000000000000000000000000000000000000000000000000000000000012"
+        );
+    }
+
+    // TODO Not sure how to set `StandardBridge public otherBridge;`
+    // L2StandardBridge's constructor doesn't take args, but calls
```

Comment:
use `mustGetAddress("L1StandardBridgeProxy");` to get the other bridge argument. The `L2StandardBridge` has an `initialize` function that should be used to set state